pkg/tasker/tasker.go

```diff
@@ -0,0 +1,309 @@
+			if err != nil {
+				return errors.Wrap(err, "tasker resubscribing to new challenges")
+			}
+		case <-mt.ctx.Done():
+			mt.done <- true
+		}
+	}
+}
+
+func (mt *MiningTasker) Stop() {
+	mt.close()
+	level.Info(mt.logger).Log("msg", "tasker shutdown complete")
+}
+
+func (mt *MiningTasker) CreateWork(challenge *tellor.ITellorNewChallenge) *mining.Work {
+	/* TODO: Do we need these anymore?
+	dispKeys := []string{}
+	for _, account := range mt.accounts {
+		dispKeys = append(dispKeys, db.DisputeStatusKeyFor(account.Address))
+
+	}
+	keys := []string{
+		db.RequestIdKey,
+		db.RequestIdKey0,
+		db.RequestIdKey1,
+		db.RequestIdKey2,
+		db.RequestIdKey3,
+		db.RequestIdKey4,
+		db.LastNewValueKey,
+	}
+	keys = append(keys, dispKeys...)
+
+	m, err := mt.proxy.BatchGet(keys)
+	if err != nil {
+		level.Error(mt.logger).Log("msg", "get data from data proxy, cannot continue at all")
+		return nil
+	}
+
+	level.Debug(mt.logger).Log("msg", "received data", "data", m)
+
+	for _, dispKey := range dispKeys {
+		if mt.checkDispute(m[dispKey]) == statusWaitNext {
+			return nil
+		}
+	}*/
+
+	diff := challenge.Difficulty
+	reqIDs := challenge.CurrentRequestId
+	for i := 0; i < 5; i++ {
+		valKey := fmt.Sprintf("%s%d", db.QueriedValuePrefix, reqIDs[i].Uint64())
+		m2, err := mt.proxy.BatchGet([]string{valKey})
+		if err != nil {
+			level.Info(mt.logger).Log(
+				"msg", "retrieve pricing data for current request id",
+				"err ", err,
+			)
+			//return nil, false
+		}
+		val := m2[valKey]
+		if len(val) == 0 {
+			if err != nil {
+				level.Error(mt.logger).Log(
+					"msg", "parsing config",
+					"err ", err,
+				)
+				return nil
+			}
+			jsonFile, err := os.Open(mt.cfg.ManualDataFile)
+			if err != nil {
+				return nil
+			}
+			defer jsonFile.Close()
+			byteValue, _ := ioutil.ReadAll(jsonFile)
+			var result map[string]map[string]uint
+			_ = json.Unmarshal([]byte(byteValue), &result)
+			_id := strconv.FormatUint(reqIDs[i].Uint64(), 10)
+			val := result[_id]["VALUE"]
+			if val == 0 {
+				level.Info(mt.logger).Log(
+					"msg", "pricing data not available for request",
+					"request", reqIDs[i].Uint64(),
+				)
+				return nil
+			}
+			level.Info(mt.logger).Log("msg", "USING MANUALLY ENTERED VALUE!!!! USE CAUTION")
+		}
+	}
+
+	newChallenge := &mining.MiningChallenge{
+		Challenge:  challenge.CurrentChallenge[:],
+		Difficulty: diff,
+		RequestIDs: reqIDs,
+	}
+
+	// If this challenge is already sent out, don't do it again.
+	if mt.currChallenge != nil &&
+		mt.instantSubmitSentForThisBlock && // Not instanst submit or instant submit but already has been sent out.
+		bytes.Equal(newChallenge.Challenge, mt.currChallenge.Challenge) { // This a new oracle block so a new challenge.
+		return nil
+	}
+
+	// When it is a new challenge reset the instant submit status.
+	if mt.currChallenge != nil && !bytes.Equal(newChallenge.Challenge, mt.currChallenge.Challenge) {
+		mt.instantSubmitSentForThisBlock = false
+	}
```

Comment:
We don't need this anymore. This was for when we needed to check for new challenge at a set interval.
```suggestion
	// If this challenge is already sent out, don't do it again.
	if mt.currChallenge != nil &&
		mt.instantSubmitSentForThisBlock && // Not instanst submit or instant submit but already has been sent out.
		bytes.Equal(newChallenge.Challenge, mt.currChallenge.Challenge) { // This a new oracle block so a new challenge.
		return nil
	}

	// When it is a new challenge reset the instant submit status.
	if mt.currChallenge != nil && !bytes.Equal(newChallenge.Challenge, mt.currChallenge.Challenge) {
		mt.instantSubmitSentForThisBlock = false
	}
```

Reply:
Done